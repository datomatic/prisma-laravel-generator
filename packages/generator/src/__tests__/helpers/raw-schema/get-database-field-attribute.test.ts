import {getRawSample} from '../../__fixtures__/get-raw-sample';
import {getSampleDmmf} from '../../__fixtures__/get-sample-dmmf';
import getDatabaseFieldAttribute from '../../../helpers/raw-schema/get-database-field-attribute';

test('getDatabaseFieldAnnotation', async () => {
  const raw = await getRawSample();
  const sampleDMMF = await getSampleDmmf();

  for (const modelInfo of sampleDMMF.datamodel.models) {
    for (const fieldInfo of modelInfo.fields) {
      expect(
        getDatabaseFieldAttribute(fieldInfo.name, modelInfo.name, raw),
      ).toMatchSnapshot(`${modelInfo.name} - ${fieldInfo.name}`);
    }
  }
});
