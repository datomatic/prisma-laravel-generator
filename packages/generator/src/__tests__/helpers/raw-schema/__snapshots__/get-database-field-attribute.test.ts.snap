// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getDatabaseFieldAnnotation: Category - id 1`] = `false`;

exports[`getDatabaseFieldAnnotation: Category - name 1`] = `false`;

exports[`getDatabaseFieldAnnotation: Category - posts 1`] = `false`;

exports[`getDatabaseFieldAnnotation: Post - author 1`] = `false`;

exports[`getDatabaseFieldAnnotation: Post - authorId 1`] = `false`;

exports[`getDatabaseFieldAnnotation: Post - categories 1`] = `false`;

exports[`getDatabaseFieldAnnotation: Post - createdAt 1`] = `false`;

exports[`getDatabaseFieldAnnotation: Post - id 1`] = `false`;

exports[`getDatabaseFieldAnnotation: Post - published 1`] = `false`;

exports[`getDatabaseFieldAnnotation: Post - title 1`] = `false`;

exports[`getDatabaseFieldAnnotation: Profile - bio 1`] = `false`;

exports[`getDatabaseFieldAnnotation: Profile - id 1`] = `false`;

exports[`getDatabaseFieldAnnotation: Profile - user 1`] = `false`;

exports[`getDatabaseFieldAnnotation: Profile - userId 1`] = `false`;

exports[`getDatabaseFieldAnnotation: User - email 1`] = `false`;

exports[`getDatabaseFieldAnnotation: User - id 1`] = `false`;

exports[`getDatabaseFieldAnnotation: User - name 1`] = `false`;

exports[`getDatabaseFieldAnnotation: User - posts 1`] = `false`;

exports[`getDatabaseFieldAnnotation: User - profile 1`] = `false`;

exports[`getDatabaseFieldAnnotation: User - role 1`] = `false`;

exports[`getDatabaseFieldAnnotation: UserWithMap - email 1`] = `"VarChar(100)"`;

exports[`getDatabaseFieldAnnotation: UserWithMap - id 1`] = `false`;

exports[`getDatabaseFieldAnnotation: UserWithMap - name 1`] = `false`;

exports[`getDatabaseFieldAnnotation: UserWithPrimaryKey - email 1`] = `false`;

exports[`getDatabaseFieldAnnotation: UserWithPrimaryKey - id 1`] = `false`;

exports[`getDatabaseFieldAnnotation: UserWithPrimaryKey - name 1`] = `false`;
